analyses/malawi/notebooks/mwi_ecmwf_monthly_skill_false_alarms.md

```diff
@@ -0,0 +1,308 @@
+df_hazl=pd.wide_to_long(df_haz,"Hazard",i=["District","TA"],j="year",sep=" ")[["start_analysis","Hazard"]].reset_index()
+#compute whether a dry spell occured
+df_hazl['dry_spell'] = df_hazl.apply(lambda x: ab_ify(x),axis=1)
+gdf_adm2=gpd.read_file(adm2_bound_path)
+df_hazl=df_hazl.merge(gdf_adm2[["ADM2_EN","ADM1_EN"]],how="left",left_on="District",right_on="ADM2_EN")
+df_mvac_year=df_hazl.groupby(["year","ADM1_EN"])["dry_spell"].agg(ds_sum="sum",ds_count="count").reset_index()
+#compute percentage of analyzed TAs experiencinga dry spell per year
+df_mvac_year["perc"]=df_mvac_year.ds_sum/df_mvac_year.ds_count*100
+# MVAC's year is the year during which thre crops were harvested, i.e. the end of the rainy season
+#While for the obs dry spells from CHIRPS we use season_approx as the year the rainy season started --> add that variable to the dataset
+df_mvac_year.rename(columns={"year":"season_approx_mvac"},inplace=True)
+df_mvac_year["season_approx"]=df_mvac_year.season_approx_mvac-1
+```
+
+```python
+df_mvac_year[df_mvac_year.ADM1_EN=="Southern"].head()
+```
+
+```python
+#plot the percentages across years (ugly just for inspection)
+df_mvac_year[df_mvac_year.ADM1_EN=="Southern"].plot(x="season_approx",y="perc",kind="bar",figsize=(5,2),legend=False,title="percentage of area experiencing a dry spell")
+```
+
+```python
+#start month of the rainy season
+start_rainy_seas=10
+#season approx indicates the year during which the rainy season started
+#this is done because it can start during one year and continue the next calendar year
+#we therefore prefer to group by rainy season instead of by calendar year
+df_forobs["season_approx"]=np.where(df_forobs.date_month.dt.month>=start_rainy_seas,df_forobs.date_month.dt.year,df_forobs.date_month.dt.year-1)
+```
+
+```python
+#group to rainy season, as mvac is yearly data
+df_forobs_seas=df_forobs.groupby(["season_approx","leadtime","ADM1_EN"],as_index=False).sum()
+```
+
+```python
+df_forobs_seas=df_forobs_seas.merge(df_mvac_year[["season_approx","ADM1_EN","perc"]],how="left",on=["season_approx","ADM1_EN"])
+df_forobs_seas.rename(columns={"perc":"perc_ds_mvac"},inplace=True)
+```
+
+```python
+#note: a neater version of computing the rp period will be implemented in utils_general/statistics.py soon
+from scipy.interpolate import interp1d
+def get_return_period_function(df,var):
+    df_rp=df.sort_values(by=var,ascending=False)
+    n=len(df)
+    #rank most extreme occurrences
+    df_rp['rank'] = np.arange(n) + 1
+    df_rp['exceedance_probability'] = df_rp['rank'] / (n+1)
+    #rp=return period
+    df_rp['rp'] = 1 / df_rp['exceedance_probability']
+    return interp1d(df_rp['rp'], df_rp[var])
+
+rp_dict = {}
+
+f_rp = get_return_period_function(df_forobs_seas.loc[df_forobs_seas.perc_ds_mvac.notnull()],"perc_ds_mvac")#[df_stats_reg.rainy_seas==1])
+for year in [1.5, 2, 3, 4, 5, 10]:
+    #round the percentages to multiples of 5 (not sure if this is too generalizing)
```

Comment:
why do we round?

Reply:
Purely for simplification. Since differences are not gonna be large (especially with this low number of observations), it was easier to communicate we are looking at a 85% than 82% threshold